python/mcap-ros2-support/mcap_ros2/writer.py

```diff
@@ -76,9 +87,10 @@ def write_message(
             type_dict = serialize_dynamic(  # type: ignore
                 schema.name, schema.data.decode()
             )
-            encoder = type_dict[schema.name]
-            if encoder is None:
+            # Check if schema.name is in type_dict
```

Comment:
fyi, you can change this line to:
```python
encoder = type_dict.get(schema.name, None)
```
and leave the rest the same, and have the correct behaviour while only searching the dictionary once. In this context it'd have negligible performance impact, so 🤷 but I figured i'd say it anyway.